skopt/tests/test_utils.py

```diff
@@ -0,0 +1,65 @@
+import numpy as np
+import tempfile
+
+from sklearn.utils.testing import assert_array_almost_equal
+from sklearn.utils.testing import assert_array_equal
+from sklearn.utils.testing import assert_array_less
+from sklearn.utils.testing import assert_equal
+from sklearn.utils.testing import assert_not_equal
+from sklearn.utils.testing import assert_less_equal
+from sklearn.utils.testing import assert_greater
+from sklearn.utils.testing import assert_true
+
+from skopt import gp_minimize
+
+from skopt import load
+from skopt import dump
+
+
+def obj_fun(x):
+    return np.sin(5 * x[0]) * (1 - np.tanh(x[0] ** 2))
+
+
+def check_optimization_results_equality(res_1, res_2):
+    # Check if the results objects have the same keys
+    assert_equal(res_1.keys(), res_2.keys())
+    # Shallow check of the main optimization results
+    assert_array_equal(res_1.x, res_2.x)
+    assert_array_equal(res_1.x_iters, res_2.x_iters)
+    assert_array_equal(res_1.fun, res_2.fun)
+    assert_array_equal(res_1.func_vals, res_2.func_vals)
+
+
+def test_dump_and_load():
+    res = gp_minimize(obj_fun,
+                      [(-2.0, 2.0)],
+                      x0=[0.],
+                      acq_func="LCB",
+                      n_calls=2,
+                      n_random_starts=0,
+                      random_state=1)
+
+    # Test normal dumping and loading
+    with tempfile.TemporaryFile() as f:
+        dump(res, f)
+        f.seek(0)
+        res_loaded = load(f)
+    check_optimization_results_equality(res, res_loaded)
+    assert_true('func' in res_loaded.specs['args'])
+
+    # Test dumping without objective function
+    with tempfile.TemporaryFile() as f:
+        dump(res, f, store_objective=False)
+        f.seek(0)
```

Comment:
Are you sure about the second point? Atleast tests pass locally after I remove the `seek` line.